streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -120,7 +118,7 @@ public void logChange(final String storeName,
         throwUnsupportedOperationExceptionIfStandby("logChange");
         // Sending null headers to changelog topics (KIP-244)
         collector.send(
-            storeToChangelogTopic.get(storeName),
+            stateManager.changelogFor(storeName),
```

Comment:
It seems this is a slight change in behavior? Before, we would only get a topic name returned in `register` was called, as we build up a local map. Now, we only rely on `stateManager` that might contain a topic name for a store, even if `register` was never called (even if the call to `super.register` call `stateManager.register` -- hence, I am wondering if we loose a guard?

Reply:
I added an exception to make sure we do not changelog not registered state stores.
I also used the registered metadata for the stores in the `send()` call. I felt that it would be cleaner.